lib/livebook/file_system/s3.ex

```diff
@@ -378,22 +417,24 @@ defimpl Livebook.FileSystem, for: Livebook.FileSystem.S3 do
       region: fields["region"],
       access_key_id: fields["access_key_id"],
       secret_access_key: fields["secret_access_key"],
-      prefix: fields["prefix"]
+      id: fields["id"],
+      hub_id: fields["hub_id"]
     })
   end
 
   def load(_file_system, fields) do
     S3.new(fields.bucket_url, fields.access_key_id, fields.secret_access_key,
       region: fields[:region],
       external_id: fields[:external_id],
-      prefix: fields[:prefix]
+      id: fields[:id],
```

Comment:
Looking at the `:id` option, at this point I think we should change `load` to just build the struct directly instead :)

We can add a migration to ensure `:region`, `:hub_id`, `:external_id` fields.

Reply:
I'll ensure in the next PR, which I will take more attention to the migration that we already have. WDYT?

Reply:
@josevalim we could but then we would need to make sure to clear all references in settings (recent files, starred files, default dir), so I think migration may be easier.

Reply:
Ah, good call. Migration it is.